cloud/defaults.go

```diff
@@ -133,19 +133,12 @@ func GenerateOSDiskName(machineName string) string {
 
 // GetDefaultImageSKUID gets the SKU ID of the image to use for the provided version of Kubernetes.
 func getDefaultImageSKUID(k8sVersion string) string {
-	if strings.HasPrefix(k8sVersion, "v1.16") {
-		return "k8s-1dot16-ubuntu-1804"
+	k8sVersion = strings.TrimPrefix(k8sVersion, "v")
```

Comment:
This is what I did in GCP to parse k8s versions: https://github.com/kubernetes-sigs/cluster-api-provider-gcp/blob/6b1aef7ac3b53c8affad389081e8551184d9273e/cloud/services/compute/instances.go#L199-L203

Reply:
thanks for the tip! changed to use the same thing